packages/client/hmi-client/src/components/workflow/tera-calibration-node.vue

```diff
@@ -1,71 +1,252 @@
+				v-for="index in calibrateNumCharts"
 				:key="index"
-				:run-results="openedWorkflowNodeStore.calibrateRunResults"
-				:run-id-list="openedWorkflowNodeStore.calibrateRunIdList"
+				:run-results="runResults"
+				:run-id-list="simulationIds"
 				:chart-idx="index"
 			/>
+			<Button
+				class="add-chart"
+				text
+				:outlined="true"
+				@click="calibrateNumCharts++"
+				label="Add Chart"
+				icon="pi pi-plus"
+			></Button>
 		</AccordionTab>
+		<!-- <AccordionTab header="Loss"></AccordionTab>
+		<AccordionTab header="Parameters"></AccordionTab>
+		<AccordionTab header="Variables"></AccordionTab> -->
 	</Accordion>
 </template>
 
 <script setup lang="ts">
-import { computed, shallowRef, watch } from 'vue';
-import { downloadRawFile, getDataset } from '@/services/dataset';
+import { computed, shallowRef, watch, ref, ComputedRef } from 'vue';
 import { WorkflowNode } from '@/types/workflow';
 import DataTable from 'primevue/datatable';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
 import Column from 'primevue/column';
 import Accordion from 'primevue/accordion';
 import AccordionTab from 'primevue/accordiontab';
+import { CalibrationRequest, CsvAsset, Simulation, ModelConfiguration } from '@/types/Types';
+import {
+	makeCalibrateJob,
+	getSimulation,
+	getRunResult
+} from '@/services/models/simulation-service';
 import { useOpenedWorkflowNodeStore } from '@/stores/opened-workflow-node';
-import { CsvAsset, Dataset } from '@/types/Types';
+import { setupModelInput, setupDatasetInput } from '@/services/calibrate-workflow';
+import { RunResults } from '@/types/SimulateConfig';
+import { csvParse } from 'd3';
+import { CalibrationOperation } from './calibrate-operation';
 import TeraSimulateChart from './tera-simulate-chart.vue';
-// import { calibrationParamExample } from '@/temp/calibrationExample';
 
 const props = defineProps<{
 	node: WorkflowNode;
 }>();
-// const modelConfig = computed(() => props.node.inputs[0].value?.[0] as ModelConfiguration | undefined);
-const datasetId = computed(() => props.node.inputs[1].value?.[0] as number | undefined);
+
+const emit = defineEmits(['append-output-port']);
 const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
-// const runId = ref('');
-// const datasetColumnNames = ref<string[]>();
+
+const modelConfigId = computed(() => props.node.inputs[0].value?.[0] as string | undefined);
+const datasetId = computed(() => props.node.inputs[1].value?.[0] as string | undefined);
+const currentDatasetFileName = ref<string>();
+const modelConfig = ref<ModelConfiguration>();
+const startedRunId = ref<string>();
+const completedRunId = ref<string>();
+
+const datasetColumnNames = ref<string[]>();
+const modelColumnNames = ref<string[] | undefined>();
+const calibrateNumCharts = ref<number>(1);
+const runResults = ref<RunResults>({});
+const simulationIds: ComputedRef<any | undefined> = computed(
+	<any | undefined>(() => props.node.outputs[0]?.value)
+);
```

Comment:
1. Don't need to address it now, but there are a few things here that do not need to be `refs` 
2. I don't understand why your `simulationIds` is so complicated ..., should it just be a `string[]`?

Reply:
Still cleaning up some stuff for sure
As for simulationIds it needs to be a ref in order for the simulation-chart to work properly - I was told by @mj3cheun 
Aside from this very fair it should not be of type any